tests/rustdoc/intra-doc/deps.rs

```diff
@@ -0,0 +1,22 @@
+// Checks that the link generated
+// Regression test for <https://github.com/rust-lang/rust/issues/137857>.
+
+//@ compile-flags: --document-private-items -Z unstable-options
+//@ compile-flags: --extern-html-root-url=empty=https://docs.rs/
+// This one is to ensure that we don't link "by default" any item we see which has
+// an external html root URL.
+//@ compile-flags: --extern-html-root-url=non_existent=https://docs.rs/
```

Comment:
```suggestion
//@ compile-flags: --document-private-items -Z unstable-options
//@ compile-flags: --extern-html-root-url=empty=https://empty.example/
// This one is to ensure that we don't link to any item we see which has
// an external html root URL unless it actually exists
//@ compile-flags: --extern-html-root-url=non_existent=https://non-existant.example/
```

I think it is confusing to give a value that is similar to a real value, but which is incorrect if you wanted the link to actually work.
Using obvious placeholder values should hopefully be more clear (`.example` in particular is reserved as a gTLD that will never actually be used)